api/controllers/WebActionsController.js

```diff
@@ -286,6 +286,54 @@ module.exports = {
           referralCode = response.body.referralCode;
         }
 
+        // GET referrer's referral count from Photon
+        //Publish 'referral' event to SNS after new user is sucessfully subscribed to MC
+        if (mcSubscribeSuccessful && req.body.referredByCode && req.body.referredByCode.length > 0) {
+
+          let referrerPhone = ReferralCodes.decode(req.body.referredByCode);
+
+          let referralCountRequest = {
+            method: 'GET',
+            uri:
+            sails.config.globals.photonApiUrl + '/referral/' + referrerPhone,
+            json: true,
+          };
+
+          request
+            .getAsync(referralCountRequest)
+            .then(resData => {
+              if (!resData || !resData.body) {
+                sails.log.error(
+                  'Invalid response received from Photon GET referral/:phone.'
+                );
+              } else {
+                sails.log.info('Successful GET referral count');
+                sails.log.info(`  referralCount: ${resData.body.referralCount}`);
+
+                let referralData = {
+                  newUser: {
+                    platform: 'sms',
+                    platformId: req.body.phone,
+                    referralCode,
+                  },
+                  referrer: {
+                    platform: 'sms',
+                    platformId: req.body.referredByCode,
```

Comment:
newUser:
-`platformId` will be the user's `id` in the `users` table. This may be `undefined` for now depending on whether or not the user has been created in db yet.
- don't need to pass req.body.phone anywhere

Referrer:
- Similarly, platformID will be the `id` of the user in the `users` table
- Add an additional property for `referralCode` which will be the `req.body.referredByCode`, ie. `xChtazl` <-- that one